### ISSUE TITLE
[BUG] Database issue on FOREIGN key

### ISSUE THREAD

**Issue by thelittlefireman** [bunkerity/bunkerweb#535]
**Description**
Database issue  on FOREIGN key after removing a custom reverse proxy
Docker v1.5.0

**How to reproduce**
- Add a new reverse proxy.
test_CUSTOM_CONF_SERVER_HTTP_auto is mandatory
- ```
      - SERVER_NAME=syno.test.test XXX.test.test
      ...
      - syno.test.test_REVERSE_PROXY_URL=/
      - syno.test.test_REVERSE_PROXY_HOST=https://192.168.1.150:5001
      - syno.test.test_REVERSE_PROXY_WS=yes
      - |
        syno.test.test_CUSTOM_CONF_SERVER_HTTP_auto-custom-syno=
        proxy_busy_buffers_size 256k;
        proxy_buffers 8 128k;
        proxy_buffer_size 128k;```

- Start bw-scheduler and bw

- remove the  reverse proxy
- ```
      - SERVER_NAME=XXX.test.test```
- restart bw-scheduler and bw.

see  bw-scheduler logs

**Logs**

```
[2023-06-25 23:11:49] - GENERATOR - ❌ - Can't save config to database : Service syno.test.test not found, please check your config
Service readarr.test.test not found, please check your config
Service home.test.test not found, please check your config
Traceback (most recent call last):
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/default.py", line 920, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/share/bunkerweb/deps/python/pymysql/cursors.py", line 158, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/pymysql/cursors.py", line 325, in _query
    conn.query(q)
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 549, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 779, in _read_query_result
    result.read()
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 1157, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 729, in _read_packet
    packet.raise_for_error()
  File "/usr/share/bunkerweb/deps/python/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/share/bunkerweb/deps/python/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.IntegrityError: (1452, 'Cannot add or update a child row: a foreign key constraint fails (`db`.`bw_custom_configs`, CONSTRAINT `bw_custom_configs_ibfk_1` FOREIGN KEY (`service_id`) REFERENCES `bw_services` (`id`) ON DELETE CASCADE ON UPDATE CASCADE)')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/share/bunkerweb/db/Database.py", line 703, in save_custom_configs
    session.commit()
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 1906, in commit
    trans.commit(_to_root=True)
  File "<string>", line 2, in commit
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/state_changes.py", line 137, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 1221, in commit
    self._prepare_impl()
  File "<string>", line 2, in _prepare_impl
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/state_changes.py", line 137, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 1196, in _prepare_impl
    self.session.flush()
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 4154, in flush
    self._flush(objects)
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 4290, in _flush
    with util.safe_reraise():
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/util/langhelpers.py", line 147, in __exit__
    raise exc_value.with_traceback(exc_tb)
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 4251, in _flush
    flush_context.execute()
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/unitofwork.py", line 467, in execute
    rec.execute(self)
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/unitofwork.py", line 644, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/persistence.py", line 93, in save_obj
    _emit_insert_statements(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/persistence.py", line 1223, in _emit_insert_statements
    result = connection.execute(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1413, in execute
    return meth(
           ^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/sql/elements.py", line 483, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1841, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 2339, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/default.py", line 920, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/share/bunkerweb/deps/python/pymysql/cursors.py", line 158, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/pymysql/cursors.py", line 325, in _query
    conn.query(q)
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 549, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 779, in _read_query_result
    result.read()
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 1157, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/pymysql/connections.py", line 729, in _read_packet
    packet.raise_for_error()
  File "/usr/share/bunkerweb/deps/python/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/share/bunkerweb/deps/python/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
sqlalchemy.exc.IntegrityError: (pymysql.err.IntegrityError) (1452, 'Cannot add or update a child row: a foreign key constraint fails (`db`.`bw_custom_configs`, CONSTRAINT `bw_custom_configs_ibfk_1` FOREIGN KEY (`service_id`) REFERENCES `bw_services` (`id`) ON DELETE CASCADE ON UPDATE CASCADE)')
[SQL: INSERT INTO bw_custom_configs (service_id, type, name, data, checksum, method) VALUES (%(service_id)s, %(type)s, %(name)s, %(data)s, %(checksum)s, %(method)s)]
[parameters: {'service_id': 'syno.test.test', 'type': 'server_http', 'name': 'auto-custom-syno', 'data': b'\nproxy_busy_buffers_size 256k;\nproxy_buffers 8 128k;\nproxy_buffer_size 128k;\n', 'checksum': '1a0dc664dd5f1223c73f3ad585a6336b9e4fd3f1ce8b13c404990168d3651719', 'method': 'scheduler'}]
(Background on this error at: https://sqlalche.me/e/20/gkpj)

```

**Comment by TheophileDiot** [comment]
Hi @thelittlefireman, thank you for opening this issue.
If I understand it well, it happens when you delete the specific domain from the setting `SERVER_NAME` but not the custom configuration ?

**Comment by TheophileDiot** [comment]
@thelittlefireman interesting, i'll try it out and let you know, thank you again

**Comment by TheophileDiot** [comment]
I just tried your issue on the dev branch and everything is in order, the changes I made to the scheduler between the 1.5.0 and the (future) 1.5.1 fixes this

**Comment by fl0ppy-d1sk** [comment]
Hello @thelittlefireman,

The fix is integrated into the 1.5.1 release, thanks again !
